main.go

```diff
@@ -50,11 +51,19 @@ func (s *server) makeGenerateQRCodeHandler() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		text := s.text
 		logoPath := s.logoPath
+		qrSize := 256
 
 		if textParam := r.URL.Query().Get("text"); textParam != "" {
 			text = textParam
 		}
 
+		qrSizeParam := r.URL.Query().Get("size")
+		size, err := strconv.Atoi(qrSizeParam)
+
+		if err == nil {
```

Comment:
This is not optimal error handling. There is no error being logged or returned if you send in a bad size.
I think that the simplest is that you return an error if the the size is not an int or has an unreasonable size (should be between 128 and 2048 or similar).

The error code should be `400 Bad Request`, I think..